src/client/metering.rs

```diff
@@ -0,0 +1,201 @@
+/// * MeteringRecorder - Use this to record all evaluations, which will eventually be sent to the server.
+pub(crate) fn start_metering<T: ServerClient>(
+    _config_id: ConfigurationId,
+    _transmit_interval: std::time::Duration,
+    server_client: T,
+) -> (MeteringTask, MeteringRecorder) {
+    let (sender, receiver) = mpsc::channel();
+
+    let thread = ThreadHandle::new(move |_terminator: mpsc::Receiver<()>| {
+        // TODO: termination handling
+        loop {
+            // TODO: error handling
+            let _ = receiver.recv().unwrap();
+            // TODO: actually process the event
+            let json_data = crate::models::MeteringDataJson {};
+            server_client.push_metering_data(&json_data);
+        }
+    });
+
+    (
+        MeteringTask {
+            _thread_handle: thread,
+        },
+        MeteringRecorder {
+            evaluation_event_sender: sender,
+        },
+    )
+}
+
+/// Allows recording of evaluation events.
+/// Communicates with the MeteringTask, which leads to eventual transmission of recorded evaluations to the server.
+pub(crate) struct MeteringRecorder {
+    evaluation_event_sender: mpsc::Sender<EvaluationEvent>,
+}
+
+impl MeteringRecorder {
+    /// Record the evaluation of a feature, for eventual transmission to the server.
+    pub fn record_feature_evaluation(
+        &self,
+        feature_id: String,
+        entity_id: String,
+        segment_id: Option<String>,
+    ) -> crate::errors::Result<()> {
+        self.evaluation_event_sender
+            .send(EvaluationEvent::Feature(EvaluationEventData {
+                subject_id: feature_id,
+                entity_id: entity_id,
+                segment_id: segment_id,
+            }))
+            .map_err(|_| crate::errors::Error::MeteringError {})
+    }
+
+    /// Record the evaluation of a property, for eventual transmission to the server.
+    pub fn record_property_evaluation(
+        &self,
+        property_id: String,
+        entity_id: String,
+        segment_id: Option<String>,
+    ) -> crate::errors::Result<()> {
+        self.evaluation_event_sender
+            .send(EvaluationEvent::Property(EvaluationEventData {
+                subject_id: property_id,
+                entity_id: entity_id,
+                segment_id: segment_id,
+            }))
+            .map_err(|_| crate::errors::Error::MeteringError {})
+    }
+}
+
+pub(crate) struct MeteringTask {
+    _thread_handle: crate::utils::ThreadHandle<()>,
+}
+
+pub(crate) struct EvaluationEventData {
+    /// ID if the subject being evaluated. E.g. feature ID.
+    pub subject_id: String,
+    /// The ID of the Entity against which the subject was evaluated.
+    pub entity_id: String,
+    /// If applicable, the segment the subject was associated to during evaluation.
+    pub segment_id: Option<String>,
+}
+
+pub(crate) enum EvaluationEvent {
+    Feature(EvaluationEventData),
+    Property(EvaluationEventData),
+}
```

Comment:
We cam do, instead:

```rust
pub(crate) enum SubjectId {
    Feature(String),
    Property(String),
}

struct EvaluationEventData {
    /// ID off the subject being evaluated. E.g. feature ID.
    pub subject_id: SubjectId,
    /// The ID of the Entity against which the subject was evaluated.
    pub entity_id: String,
    /// If applicable, the segment the subject was associated to during evaluation.
    pub segment_id: Option<String>,
}
```

and maybe we can have only function merging `record_property_evaluation` and `record_feature_evaluation`

```rust
pub fn record_evaluation(
        &self,
        subject_id: SubjectId,
        entity_id: String,
        segment_id: Option<String>,
    ) -> crate::errors::Result<()> {
```